### PR TITLE
paw3902 minor improvements

### PR DESCRIPTION
**Pull request by dagar**
- verify register writes
 - cleanly reset on lighting mode changes as per the datasheet
 - perf counters for mode switches and other errors
 - cleanly reset on false motion detection
 - minimize sleeps
 - allowing setting custom rotation yaw angle directly on command line with -Y (ignoring SENS_FLOW_ROT)
 - enable LED output

### CODE REVIEW

File: src/drivers/optical_flow/paw3902/PAW3902.cpp [hunk 1]
```diff
@@ -633,68 +694,74 @@ PAW3902::RunImpl()
...
 	// returns if the collect time has not been reached
-	if (_flow_dt_sum_usec < _collect_time) {
-		perf_end(_sample_perf);
+	if (_flow_dt_sum_usec < COLLECT_TIME) {
 		return;
 	}
 
 	optical_flow_s report{};
 	report.timestamp = timestamp_sample;
+	//report.device_id = get_device_id();
```
Comment (mhkabir) [inline src/drivers/optical_flow/paw3902/PAW3902.cpp:721]: Why not?

File: src/drivers/optical_flow/paw3902/PAW3902.cpp [hunk 2]
```diff
@@ -633,68 +694,74 @@ PAW3902::RunImpl()
...
 
-	/* No gyro on this board */
+	// No gyro on this board
 	report.gyro_x_rate_integral = NAN;
 	report.gyro_y_rate_integral = NAN;
 	report.gyro_z_rate_integral = NAN;
 
 	// set (conservative) specs according to datasheet
-	report.max_flow_rate = 5.0f;       // Datasheet: 7.4 rad/s
+	report.max_flow_rate = 7.4f;        // Datasheet: 7.4 rad/s
```
Comment (mhkabir) [inline src/drivers/optical_flow/paw3902/PAW3902.cpp:742]: I believe this was set more conservatively for the PMW3901 after real world testing.

File: src/drivers/optical_flow/paw3902/PAW3902.cpp [hunk 3]
```diff
@@ -633,68 +694,74 @@ PAW3902::RunImpl()
...
 	report.min_ground_distance = 0.08f; // Datasheet: 80mm
 	report.max_ground_distance = 30.0f; // Datasheet: infinity
 
+	report.timestamp = hrt_absolute_time();
```
Comment (mhkabir) [inline src/drivers/optical_flow/paw3902/PAW3902.cpp:746]: This is set twice now (previously on line 720)

File: src/drivers/optical_flow/paw3902/paw3902_main.cpp [hunk 4]
```diff
@@ -65,19 +62,18 @@ I2CSPIDriverBase *PAW3902::instantiate(const BusCLIArguments &cli, const BusInst
...
-	int ch;
+	int ch = 0;
 	using ThisDriver = PAW3902;
 	BusCLIArguments cli{false, true};
 	cli.spi_mode = SPIDEV_MODE0;
-	cli.default_spi_frequency = PAW3902_SPI_BUS_SPEED;
+	cli.default_spi_frequency = SPI_SPEED;
 
-	while ((ch = cli.getopt(argc, argv, "R:")) != EOF) {
+	while ((ch = cli.getopt(argc, argv, "Y:")) != EOF) {
```
Comment (mhkabir) [inline src/drivers/optical_flow/paw3902/paw3902_main.cpp:73]: Is there any benefit to not keeping the original `-R` convention used in all other drivers? I'm also thinking of future use if/when we support arbitrary flow sensor orientations.

**Comment by mhkabir** [comment]
I'll give this a flight tomorrow.

File: src/drivers/optical_flow/paw3902/PAW3902.cpp (hunk 1 above)
Comment (dagar) [reply src/drivers/optical_flow/paw3902/PAW3902.cpp:721]: The message hasn't been updated yet.

File: src/drivers/optical_flow/paw3902/PAW3902.cpp (hunk 3 above)
Comment (dagar) [reply src/drivers/optical_flow/paw3902/PAW3902.cpp:746]: This was pulled out of a branch where optical flow was updated to be more in line with other sensor publications.

 - sensor_optical_flow.msg
     - uint32_t device_id
     - timestamp (uORB publication metadata)
     - timestamp_sample (time corresponding as close as possible to the raw data)

I suppose I should keep the other timestamp for now until the reset is updated.

File: src/drivers/optical_flow/paw3902/paw3902_main.cpp (hunk 4 above)
Comment (dagar) [reply src/drivers/optical_flow/paw3902/paw3902_main.cpp:73]: Not that I could think of. 

I'm hoping we can make this all configurable per instance "soon". https://github.com/PX4/PX4-Autopilot/issues/15683

Comment (mhkabir) [reply src/drivers/optical_flow/paw3902/paw3902_main.cpp:73]: Then let's revert this to keep in sync with the existing style.

Comment (dagar) [reply src/drivers/optical_flow/paw3902/paw3902_main.cpp:73]: Oh sorry, I misread your comment as "is there any benefit to keeping the original -R".

1. We currently only support flow facing down.
2. People are doing things like putting these on the arm of a multicopter which might not be at a 45 degree multiple.

Comment (mhkabir) [reply src/drivers/optical_flow/paw3902/paw3902_main.cpp:73]: Ah, kk.

**Comment by dagar** [comment]
This has been flight tested quite a few times. Let's keep going, I still want to do a small follow up that gets rid of the remaining sleeps.

**Pull request merged**